File: chattools/src/main/java/aytackydln/chattools/telegram/dto/models/Updates.java
package aytackydln.chattools.telegram.dto.models;

import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import com.fasterxml.jackson.databind.util.ArrayIterator;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.util.Iterator;

@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
@Data
@RequiredArgsConstructor
public class Updates implements Iterable<Update>{
	private boolean ok;
	private Update[] result;

	@Override
	public Iterator<Update> iterator(){
		return new ArrayIterator<>(result);
	}
}

File: chattools/src/main/kotlin/aytackydln/chattools/telegram/dto/models/Updates.kt
package aytackydln.chattools.telegram.dto.models

import com.fasterxml.jackson.databind.PropertyNamingStrategies
import com.fasterxml.jackson.databind.annotation.JsonNaming

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
data class Updates(
    val result: List<Update>,
    val ok: Boolean = false,
) : Iterable<Update> {

    override fun iterator(): Iterator<Update> {
        return result.iterator()
    }
}
